### DISCUSSION TITLE
Is it possible to run backend tests locally without wiping out my local database?

### DISCUSSION THREAD

**Discussion by gpspake** [fastapi/full-stack-fastapi-template#1137] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
no code
```

### Description

Running tests locally in the way described and recommended by the docs in the readme wipes out all the data from my database (running via docker compose)

- Create the project using `copier` [per readme](https://github.com/tiangolo/full-stack-fastapi-template/blob/master/backend/README.md#local-development)
- Run the app locally via `docker compose up` [per readme](https://github.com/tiangolo/full-stack-fastapi-template/blob/master/backend/README.md#local-development) - At this point everything works and you can add items to the db via the client app or the Open Api doc. You can connect to the local db (running in docker) and see the changes.
- Run tests via `bash ./scripts/test.sh` [per readme](https://github.com/tiangolo/full-stack-fastapi-template/blob/master/backend/README.md#backend-tests)

When you run docker compose up again, you'll notice that all the records you created are gone because the tests use the same db. 

### Operating System

macOS

### Operating System Details

Macbook M1

### Python Version

3.10

**Comment by gpspake** [comment]
I think the bottom line here is that local tests and development - both in accordance with the usage described in the Readme docs - use the same db. So you can't work with the app locally and run tests without wiping out all your progress. It seems like there's no way I'm the only one who's encountered this and it makes local dev kind of untenable so I'm hoping I'm just overlooking some obvious config of where I need to be setting `POSTGRES_DB` or whatever.

**Reply by swelborn** [reply]
adding semi-related backend test issue for more exposure #1135 

the tests fail (i think) because the DB is altered during `generate_password_reset_token`. probably best to set up a test db as you mention...
